File: 백준/Silver/5426. 비밀 편지/비밀 편지.py
import math

a = int(input())
for _ in range(a):
  ip = list(input())
  length = int(math.sqrt(len(ip)))

  for i in range(length - 1, -1, -1):
    for j in range(length):
      print(ip[i + j * length], end = '')
  
  print()

File: 백준/Silver/5426. 비밀 편지/비밀 편지.c
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void) { 
   int a;
   scanf("%d", &a);
   for(int k = 0; k < a; k++){
      char string[10001];
      
      scanf("%s", string);

      int len = strlen(string);
      int count = sqrt(len);

      char new_str[count][count + 1];
      for(int i = 0; i < count; i++){
         for(int j = 0; j < count; j++){
            new_str[i][j] = string[i * count + j];
         }
         new_str[i][count] = '\0';
      }
      for(int i = count - 1; i > -1; i--){
         for(int j = 0; j < count; j++){
            printf("%c", new_str[j][i]);
         }
      }
      printf("\n");
   }

   return 0;
}